components/capi/src/pluralrules.rs

```diff
@@ -0,0 +1,129 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locid::Locale as ICULocale;
+use icu_plurals::{PluralCategory, PluralOperands, PluralRuleType};
+
+
+use crate::provider::ICU4XErasedDataProvider;
+
+use std::ptr;
+
+// opaque type
+pub use icu_plurals::PluralRules as ICU4XPluralRules;
+
+#[repr(C)]
+pub struct ICU4XCreatePluralRulesResult {
+    rules: *mut ICU4XPluralRules,
+    success: bool,
+}
+
+#[no_mangle]
+pub extern "C" fn icu4x_plural_rules_create(
+    locale: &ICULocale,
+    provider: &mut ICU4XErasedDataProvider,
```

Comment:
Can you take `provider` as non-`mut`?

Reply:
Oh i could swear it needed it as mut. Fixing